tests/trnsysGUI/connection/testCleanConnector.py

```diff
@@ -0,0 +1,221 @@
+import unittest.mock as _m
+
+import trnsysGUI.connection.getNiceConnector as _gnc
+from trnsysGUI import cornerItem as _ci
+
+from . import fakeConnections as _fc
+
+spConnection = _fc.createSimpleSPConnectionMock("spConnectionName")
+
+
+class FakeSegmentItem(_fc.StrictMockBase):
+    def __init__(self, startNode, endNode, connection, **kwargs):
+        super().__init__(**kwargs)
+        self.startNode = startNode
+        self.endNode = endNode
+        self.connection = connection
+        self.lineStart = None
+        self.lineEnd = None
+
+    def setLine(self, point1, point2):
+        self.lineStart = point1
+        self.lineEnd = point2
+
+
+class FakeSegmentItemFactory:
+    def __init__(self, connection):
+        self._connection = connection
+
+    def create(self, startNode, endNode):
+        return FakeSegmentItem(startNode, endNode, self._connection)
+
+
+class TestCleanConnector:
+    def testDPconnectionUsingBothTwoConnector(self):
+        self._runNiceConnWithTests(
+            connectorType=_gnc.NiceConnectorBothTwo,
+            patchString="NiceConnectorBothTwo",
+            addItemCount=1,
+            fromPortSide=2,
+            toPortSide=2,
+            nrOfNewItems=9,
+            nSegsExpected=5,
+            xStartCoordsExpected=[0.0, 30.0, 30.0, 130.0, 130.0],
+            xEndCoordsExpected=[30.0, 30.0, 130.0, 130.0, 100.0],
+            yStartCoordsExpected=[0.0, 0.0, 200.6, 200.6, 100.0],
+            yEndCoordsExpected=[0.0, 200.6, 200.6, 100.0, 100.0],
+        )
+
+    def testDPconnectionUsingBothZeroConnector(self):
+        self._runNiceConnWithTests(
+            patchString="NiceConnectorBothZero",
+            connectorType=_gnc.NiceConnectorBothZero,
+            addItemCount=1,
+            fromPortSide=0,
```

Comment:
use clearer descriptions for the port sides, here:
"left, right, top, bottom"
adjust this to numbers elsewhere?
with enums!

Reply:
Then, the tests could be called "bothLeft" etc.